src/context/sessionContext/index.test.tsx

```diff
@@ -167,6 +174,46 @@ describe("SessionContext functionality", () => {
     });
   });
 
+  it("fetches the user's profile if logging in succeeds", async () => {
+    (handleIncomingRedirect as jest.Mock).mockResolvedValueOnce({
+      webId: "https://some.webid",
+    });
+
+    const mockedClientModule = jest.requireMock("@inrupt/solid-client");
+    const actualClientModule = jest.requireActual("@inrupt/solid-client");
+    mockedClientModule.getProfileAll = jest.fn().mockResolvedValue({
+      webIdProfile:
+        actualClientModule.mockSolidDatasetFrom("https://some.webid"),
+      altProfileAll: [
+        actualClientModule.mockSolidDatasetFrom("https://some.profile"),
+      ],
+    });
+
+    const session = {
+      info: {
+        isLoggedIn: true,
+        webId: "https://fakeurl.com/me",
+      },
+      on: jest.fn(),
+    } as any;
+
+    (getDefaultSession as jest.Mock).mockReturnValue(session);
+
+    const documentBody = render(
+      <SessionProvider sessionId="key">
+        <ChildComponent />
+      </SessionProvider>
+    );
+    await waitFor(async () => {
+      expect(mockedClientModule.getProfileAll).toHaveBeenCalled();
+      let testElement = await documentBody.findAllByTestId("profile");
+      expect(testElement[0].textContent).toBe("1 alt profiles found");
```

Comment:
A perhaps cleaner way of writing these assertions is:

```suggestion
      expect(documentBody.getByTestId("profile")).toHaveTextContent("1 alt profiles found");
```

Using this format, you skip the additional lets, and it also errors more nicely if the element is not found within the screen. (Aside: I think normally it's standard to call the result of `render()` from RTL as "screen" instead of "documentBody", also worth noting it's actually not a `document.body`).

You could also move most of the assertions outside of the waitFor, and just do:

```
await waitFor(async () => documentBody.getByTestId("profile"));
```